pkg/cliplugins/workload/plugin/workload.go

```diff
@@ -0,0 +1,123 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package plugin
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+
+	kcpclientset "github.com/kcp-dev/kcp/pkg/client/clientset/versioned"
+)
+
+// Cordon the workload cluster and mark it as unschedulable
+func (c *Config) Cordon(ctx context.Context, workloadClusterName string) error {
+	config, err := clientcmd.NewDefaultClientConfig(*c.startingConfig, c.overrides).ClientConfig()
+	if err != nil {
+		return err
+	}
+
+	err = modifyCordon(ctx, config, workloadClusterName, true)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Uncordon the workload cluster and mark it as schedulable
+func (c *Config) Uncordon(ctx context.Context, workloadClusterName string) error {
+	config, err := clientcmd.NewDefaultClientConfig(*c.startingConfig, c.overrides).ClientConfig()
+	if err != nil {
+		return err
+	}
+
+	err = modifyCordon(ctx, config, workloadClusterName, false)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// change the workload cluster cordon value
+func modifyCordon(ctx context.Context, config *rest.Config, workloadClusterName string, cordon bool) error {
+	kcpClient, err := kcpclientset.NewForConfig(config)
+	if err != nil {
+		return fmt.Errorf("failed to create kcp client: %w", err)
+	}
+
+	workloadCluster, err := kcpClient.WorkloadV1alpha1().WorkloadClusters().Get(ctx,
+		workloadClusterName,
+		metav1.GetOptions{},
+	)
+	if err != nil {
+		return fmt.Errorf("failed to get workloadcluster %s: %w", workloadClusterName, err)
+	}
+
+	workloadCluster.Spec.Unschedulable = cordon
+	// if uncordon, make sure evictAfter is not set
+	if !cordon {
+		workloadCluster.Spec.EvictAfter = nil
+	}
+
+	_, err = kcpClient.WorkloadV1alpha1().WorkloadClusters().Update(ctx, workloadCluster, metav1.UpdateOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to update workloadcluster %s: %w", workloadClusterName, err)
+	}
+
+	return nil
+}
+
+// Uncordon the workload cluster and mark it as schedulable
+func (c *Config) Drain(ctx context.Context, workloadClusterName string) error {
+	config, err := clientcmd.NewDefaultClientConfig(*c.startingConfig, c.overrides).ClientConfig()
+	if err != nil {
+		return err
+	}
+
+	kcpClient, err := kcpclientset.NewForConfig(config)
+	if err != nil {
+		return fmt.Errorf("failed to create kcp client: %w", err)
+	}
+
+	workloadCluster, err := kcpClient.WorkloadV1alpha1().WorkloadClusters().Get(ctx,
+		workloadClusterName,
+		metav1.GetOptions{},
+	)
+	if err != nil {
+		return fmt.Errorf("failed to get workloadcluster %s: %w", workloadClusterName, err)
+	}
+
+	nowTime := metav1.NewTime(time.Now())
+	workloadCluster.Spec.EvictAfter = &nowTime
```

Comment:
Do you think it's useful to let the user optionally specify a time in the future, e.g "drain in 30 minutes"?

Reply:
At a future point, if a customer really needs it, a time delay option might be able to be added.  But right now, I don't see if being needed.   There are various aspects of drain for pod that cannot be done for workloadcluster.  Most importantly the ability to have it be sync so that when it returns, the drain is complete.  I think there would need to be additional status added to workloadcluster in order for drain to know when to return.